lib/utils/index.ts

```diff
@@ -257,7 +257,11 @@ export function parseURL(url) {
 
   const result: any = {};
   if (parsed.auth) {
-    result.password = parsed.auth.split(":")[1];
+    const parsedAuth = parsed.auth.split(":");
+    if (parsedAuth[0].length > 0) {
```

Comment:
```suggestion
    if (parsedAuth[0]) {
```

I think `parsedAuth[0]` is sufficient to detect empty strings.

Reply:
The reason I used `.length` rather than `parsedAuth[0]` only is because I've been bitten before by the issue where the string `"0"` can be coerced to `0` and thus become falsey. I just tested it and it does seem like within the `if()`, `"0"` is also considered truthy, so this suggestion is valid.

https://developer.mozilla.org/en-US/docs/Glossary/truthy